File: 2352-design-an-atm-machine/2352-design-an-atm-machine.java
class ATM {
    private int[] notes = new int[5];
    private final int[] values = {20, 50, 100, 200, 500};

    public ATM() {
        
        
    }
    
    public void deposit(int[] banknotesCount) {
        for(int i = 0; i < notes.length; i++){
            notes[i] += banknotesCount[i];
        }


        
    }
    
    public int[] withdraw(int amount) {
        int[] used = new int[5];
        // int remaining = amount;
        // for(int i = 4 ; i>=0 ; i--){
        //     int available = notes[i];
        //     int use = Math.min(remaining/values[i],available);
        //     used[i] = (int)use;
        //     remaining -= use * values[i];

        // }
        int index = 4;
        while(amount > 0 && index >= 0){
            int takethismany = Math.min(amount/values[index], notes[index]);
            used[index] = takethismany;
            amount -= takethismany * values[index];
            index--;
        }
        if(amount != 0){
            return new int[]{-1};
        }
        for(int j = 0 ; j<5 ; j++){
            notes[j] -= used[j];
        }
        return used;

        
    }
}

/**
 * Your ATM object will be instantiated and called as such:
 * ATM obj = new ATM();
 * obj.deposit(banknotesCount);
 * int[] param_2 = obj.withdraw(amount);
 */

File: 2352-design-an-atm-machine/2352-design-an-atm-machine.py
class ATM:
    

    def __init__(self):
        self.notes = [0]*5
        self.values = [20,50,100,200,500]
        

    def deposit(self, banknotesCount: List[int]) -> None:
        for i in range(len(banknotesCount)) :
            self.notes[i] += banknotesCount[i]


        

    def withdraw(self, amount: int) -> List[int]:
        res = [0]*5
        remaining = amount
        for i in range(4,-1,-1) :
            available = self.notes[i]
            use = min(remaining//self.values[i],available)
            res[i] = (use)
            remaining -= use * self.values[i]
        if remaining != 0 :
            return [-1]
        for j in range(5) :
            self.notes[j] -= res[j]
        return res

        


# Your ATM object will be instantiated and called as such:
# obj = ATM()
# obj.deposit(banknotesCount)
# param_2 = obj.withdraw(amount)